manifests/prepare/package.pp

```diff
@@ -5,21 +5,31 @@
 # @param source
 #   The source file for the puppet-agent package. Can use any of the data types
 #   and protocols that the File resource's source attribute can.
+# @param destination_name
+#   The destination file name for the puppet-agent package. If no destination
+#   is given, then the basename component of the source will be used as the
+#   destination name.
 class puppet_agent::prepare::package (
   Variant[String, Array] $source,
+  Optional[String] $destination_name = undef
```

Comment:
Are empty strings allowed here, or should be prohibit them?
```suggestion
  Optional[String[1]] $destination_name = undef
```

Reply:
Moved to Windows PR